### PR TITLE
DLC Wallet & GUI touch ups

### LINKED ISSUE #2283: Make first and last textbox on DLC GUI be 0 and 2^numdigits-1
> I think I have an idea, we should have the first and last textbox have x-coord read only and set to 0 and base^numdigits - 1 and then add rows in between

From @nkohen on slack 

I messed this up when filling out coodinates on our DLC, and we should really default textboxs for the two edge cases on this screen

Screenshot from 2020-11-20 13-54-33

### LINKED ISSUE #2284: Rename `Collateral` in the GUI when creating a DLC offer
@nkohen hinted that this will need to refactor backend code as well, but `Collateral` is very misleading when constructing an offer, as it should just be _Your collateral_ 

Screenshot from 2020-11-20 14-40-06

### LINKED ISSUE #2301: Make a 'execute' option when viewing DLCs from the GUI
Currently we have 'view', we should probably have 'execute' as well. Then you can pre-populate fields like contract info

Screenshot from 2020-11-30 14-28-18

### LINKED ISSUE #2302: Blinking cursor on Contract ID and Oracle Signature textbox simultaneously
When I open `Execute` dialog I have a blinking cursor in _both_ textboxs 

When I

1. Click into the `ContractID` textbox
2. Click in to the `Oracle Signature` textbox
3. Press delete

I am deleting text inside of `ContractID` textbox when i've clicked into the `Oracle Signature` textbox. I'm guessing these two things are related.

Screenshot from 2020-11-30 16-11-34

### LINKED ISSUE #2307: Hide Demo Oracle on Mainnet (and maybe Testnet too)
Currently a demo oracle is created anytime you create a contract info on the DLC GUI and it caches the corresponding oracle info and populates it in the offer dialog, so that if a user forgets to paste in the oracle they are using it doesn't fail and instead uses the demo oracle ... which we definitely don't want happening on mannet.

### PR DESCRIPTION
**Pull request by benthecarman**
Now purges DLC sigs after execution. Because of this removed DLCStatus because it would be unavailable after execution, however, we store all needed data in the database (sigs and outcome).

Also does a bunch of gui updates:
Fixes #2302 
Fixes #2301 
Fixes #2284 
Fixes #2283
Fixes #2307

### CODE REVIEW

File: app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala [hunk 1]
```diff
@@ -1,242 +1,263 @@
...
+import org.bitcoins.commons.jsonmodels.dlc.{DLCState, DLCStatus}
+import org.bitcoins.testkit.core.gen.{CryptoGenerators, NumberGenerator, TLVGen}
 import org.bitcoins.testkit.util.BitcoinSAsyncTest
 import org.scalacheck.Gen
 
-import scala.concurrent.Future
-
 class DLCStatusTest extends BitcoinSAsyncTest {
-  behavior of "DLCStatus"
+  behavior of "SerializedDLCStatus"
```
Comment (nkohen) [inline app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala]: ping

Comment (nkohen) [reply app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala]: ditto with all the test names below

File: app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala [hunk 2]
```diff
@@ -1,242 +1,263 @@
...
 
-        val outcomeIndex = outcomeNum % offer.contractInfo.allOutcomes.length
-        val outcome = offer.contractInfo.allOutcomes(outcomeIndex.toInt)
-        val oracleSig =
-          oracleKey.schnorrSignWithNonce(
-            CryptoUtil.sha256(outcome.serialized.head).bytes,
-            kValue)
+        val totalCollateral = offer.contractInfo.max
+
+        val rand = Math.random() * offer.contractInfo.allOutcomes.size
```
Comment (nkohen) [inline app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala]: nit: `scala.util.Random.nextInt(_)`

File: app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala [hunk 3]
```diff
@@ -1,242 +1,263 @@
...
-                                    closingTx)
-
-          assert(status.oracleSig == oracleSig)
-          assert(status.outcome == outcome)
-          assert(DLCStatus.fromJson(status.toJson) == status)
-        }
+
+        val totalCollateral = offer.contractInfo.max
+
+        val rand = Math.random() * offer.contractInfo.allOutcomes.size
```
Comment (nkohen) [inline app-commons-test/src/test/scala/org/bitcoins/commons/dlc/DLCStatusTest.scala]: ditto

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/BitcoinSServerInfo.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package org.bitcoins.commons.jsonmodels
+
+import org.bitcoins.core.config.{BitcoinNetwork, BitcoinNetworks}
+import org.bitcoins.crypto.DoubleSha256DigestBE
+import ujson._
+
+case class BitcoinSServerInfo(
```
Comment (nkohen) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/BitcoinSServerInfo.scala:8]: scaladoc

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 5]
```diff
@@ -147,20 +150,20 @@ object ConsoleCli {
       cmd("createdlcoffer")
         .action((_, conf) =>
           conf.copy(
-            command = CreateDLCOffer(OracleInfo.dummy,
+            command = CreateDLCOffer(null,
```
Comment (nkohen) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: can we not use a dummy value for announcements?

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala [hunk 6]
```diff
@@ -14,22 +17,57 @@ object AcceptDLCDialog
...
+    val fromAnnouncement = OracleInfo.fromOracleAnnouncement(announcement)
+
+    fromOffer == fromAnnouncement
+  }
+
   override def constructFromInput(
       inputs: Map[String, Node]): AcceptDLCCliCommand = {
     offerDLCFile match {
       case Some(file) =>
+        // TODO figure how to validate when using a file
```
Comment (nkohen) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala:43]: ping and/or issue

File: dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCExecutionTest.scala [hunk 7]
```diff
@@ -6,10 +6,10 @@ import org.bitcoins.commons.jsonmodels.dlc.DLCMessage.{
   SingleNonceContractInfo
 }
 import org.bitcoins.commons.jsonmodels.dlc.DLCState
-import org.bitcoins.commons.jsonmodels.dlc.DLCStatus.{
-  Claimed,
-  Refunded,
-  RemoteClaimed
+import org.bitcoins.commons.jsonmodels.dlc.SerializedDLCStatus.{
```
Comment (nkohen) [inline dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCExecutionTest.scala]: Didn't this get renamed?

File: dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCMultiNonceExecutionTest.scala [hunk 8]
```diff
@@ -6,7 +6,10 @@ import org.bitcoins.commons.jsonmodels.dlc.DLCMessage.{
   SingleNonceContractInfo
 }
 import org.bitcoins.commons.jsonmodels.dlc.DLCState
-import org.bitcoins.commons.jsonmodels.dlc.DLCStatus.{Claimed, RemoteClaimed}
+import org.bitcoins.commons.jsonmodels.dlc.SerializedDLCStatus.{
```
Comment (nkohen) [inline dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCMultiNonceExecutionTest.scala]: ditto

File: dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala [hunk 9]
```diff
@@ -216,6 +215,23 @@ abstract class DLCWallet extends Wallet with AnyDLCHDWalletApi {
...
+            for {
+              updated <- updatedF
+
+              _ <- {
+                updated.state match {
+                  case DLCState.Claimed | DLCState.RemoteClaimed |
+                      DLCState.Refunded =>
+                    logger.info(
+                      s"Deleting unneeded DLC signatures for contract ${updated.contractIdOpt.get}")
+                    dlcSigsDAO.deleteByParamHash(updated.paramHash)
```
Comment (nkohen) [inline dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala:237]: might be worth sanity checking sigs and outcome are present in the db before wiping? Also is there any chance we want to put a timer on this happening (e.g. delete after 10 days or something like that?)

**Review by nkohen — COMMENTED** [review]
Looking good, just a couple of not too large things and also it would be nice if you commented on this PR if/once you have run this end-to-end with the GUI on Regtest using announcement and getting info at each step and such

File: dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCExecutionTest.scala (hunk 7 above)
Comment (nkohen) [reply dlc-wallet-test/src/test/scala/org/bitcoins/dlc/wallet/DLCExecutionTest.scala]: Yup, this is causing CI failure: https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/747088876#L472

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala (hunk 6 above)
Comment (benthecarman) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/AcceptDLCDialog.scala:43]: #2311

**Review by nkohen — APPROVED** [review]
ACK on CI and local testing

**Pull request merged**